validators/self-recharging-validator-wallets.adoc

```diff
@@ -0,0 +1,139 @@
+|===
+
+== Replenishment
+
+Self-recharging validator wallets can be replenished in ETH by anyone with the wallet's contract address and the `ValidatorId` or `sChainId`. This functionality allows anyone to ensure the continued ETH funding and operation of validator nodes and SKALE Chains.
+
+If the *self-recharging*...
+
+* *validator balance* runs out of ETH, then the nodes' wallets are first spent until transactions revert.
+* *SKALE Chain Owner balance* runs out of ETH, then transactions will revert.
+
+Node wallets connected to the self-recharging validator wallet is replenished automatically at the end of each transaction.
+
+[IMPORTANT]
+If the validator self-recharging wallet is out of ETH then node wallets start using their own ETH until the node balance is 0. Suppose the validator self-recharging wallet has 0 balance at the same time that the node wallet balance is 0. In that case, validators are then responsible for recharging both the validator self-recharging wallet and the node wallets with 0 balance.
+
+[WARNING]
+If a node's getBounty transaction reverts and can't complete the transaction before the end of the month, the validator-delegators will receive less bounty. Validators must ensure sufficient balance and a working endpoint to connect with Ethereum mainnet.
+
+[NOTE]
+There is no easy way to know the EVM's refund counter's value from inside the EVM. Thus, refunds may offset the transaction's final net cost, and node balances may slightly increase over time. This process will be reviewed and improved over time.
+
+== Using a self-recharging wallet
+
+When a new validator is registered, a self-recharging wallet is automatically deployed. The validator's private key has withdraw access to this wallet. Validators can use this private key to close the self-recharging wallet should the validator leave the network.
+
+As a validator links nodes to their validator ID, these linked nodes will automatically request reimbursement from the validator's self-recharging wallet for those transactions. If a node is unlinked from the validator, this node will no longer request reimbursement from the validator's self-recharging wallet.
+
+Existing validators will automatically be added to the Wallet contract and will start using them once this feature is deployed to mainnet.
+
+== Interacting with the Self-Recharging Wallet (SRW)
+
+Validators can use the CLI to recharge, withdraw funds from, or check the self-recharging wallet's balance. Also anyone can recharge or check the balance by interacting with the Wallet contract on Ethereum.
+
+For the CLI::
+
+```
+# Recharges the self-recharging wallet from the validator wallet (sk-val wallet info)
+sk-val srw recharge AMOUNT_IN_ETH [--gas-price PRICE_IN_GWEI]
+
+> sk-val srw recharge 0.1 --pk-file ./tests/test-pk.txt
+
+# Withdraw an amount from the self-recharging wallet
+sk-val srw withdraw [AMOUNT_IN_ETH] [--pk-file PATH_TO_PK] [--gas-price PRICE_IN_GWEI]
```

Comment:
This only withdraws from a self-recharging wallet, If any validator wants to exit their node or withdraw from now they are still responsible to withdraw from their node account by using the `skale node withdraw right?

Reply:
beautiful!